alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportIndexMonitorAction.kt

```diff
@@ -400,6 +400,19 @@ class TransportIndexMonitorAction @Inject constructor(
         private suspend fun indexMonitor() {
             var metadata = createMetadata()
 
+            if (user != null) {
+                // Use the backend roles which is an intersection of the requested backend roles and the user's backend roles.
+                // Admins can pass in any backend role. Also if no backend role is passed in, all the user's backend roles are used.
+                val rbacRoles = if (request.rbacRoles.isNullOrEmpty()) user.backendRoles.toSet()
+                else if (!isAdmin(user)) request.rbacRoles?.intersect(user.backendRoles)?.toSet()
+                else request.rbacRoles
+
```

Comment:
we should throw an error if `rbacRoles.length != request.rbacRoles` if `request.rbacRoles` is not null. This is valid for second case.

If  `request.rbacRoles` is not null but empty then for non-admins we should keep the `rbacRoles` empty.

Reply:
If a non-admin can expunge all roles from their own Monitors, that means only the admin can fix it. Should we allow for that? It might cause noise for the admin due to user error. What would be the use case to lose access to something for a non-admin user?

Reply:
I see where you're coming from but non-admin removing all users they have access to will essentially lock away the object from them in RBAC cases unless an admin gives them access again. What if a user uses the empty list and causes everyone to lose access to it since they removed backend roles that were shared by everyone? It seems like a mechanic only admin should have. I'm just concerned about the operational implications it can have to users who don't understand its implications.